host-controller/src/main/java/org/jboss/as/host/controller/Main.java

```diff
@@ -130,14 +130,16 @@ private HostControllerBootstrap boot(String[] args, final String authCode) {
         try {
             // TODO make this settable via an embedding process
             final long startTime = Module.getStartTime();
-            final HostControllerEnvironment config = determineEnvironment(args, startTime);
-            if (config == null) {
+            final HostControllerEnvironmentWrapper hostControllerEnvironmentWrapper = determineEnvironment(args, startTime);
+            if (hostControllerEnvironmentWrapper.getHostControllerEnvironment() == null) {
                 usage(); // In case there was an error determining the environment print the usage
-                abort();
+                if (hostControllerEnvironmentWrapper.getHostControllerEnvironmentStatus() == HostControllerEnvironmentWrapper.HostControllerEnvironmentStatus.ERROR) {
+                    abort();
+                }
```

Comment:
We need an 'else' clause here where we still end up calling SystemExiter.logAndExit(), just with an ExitLogger that's a no-op. Might as well create one as a constant in the SystemExiter class.